libraries/kotlin.test/common/src/main/kotlin/kotlin/test/Assertions.kt

```diff
@@ -72,6 +72,21 @@ fun <@OnlyInputTypes T> assertNotSame(illegal: T, actual: T, message: String? =
     asserter.assertNotSame(message, illegal, actual)
 }
 
+/** Asserts that [value] is [T], with an optional [message]. */
+@SinceKotlin("1.4")
+@InlineOnly
+inline fun <reified T: Any> assertIs(value: Any, message: String? = null) {
+    contract { returns() implies (value is T) }
+    asserter.assertTrue({ messagePrefix(message) + "Expected value to be of type <${T::class}>, actual <${value::class}>." }, value is T)
+}
+
+/** Asserts that [value] !is [T], with an optional [message]. */
+@SinceKotlin("1.4")
+@InlineOnly
+inline fun <reified T: Any> assertNotIs(value: Any, message: String? = null) {
+    asserter.assertFalse({ messagePrefix(message) + "Expected value to not be of type <${T::class}>" }, value is T)
```

Comment:
`messagePrefix` is internal and needs a `@PublisedApi` annotation to be used in an inline function.

Reply:
Would you recommend that? I could also add an internal non-inline helper function that calls it. What do you think?

Reply:
Oh never mind. That's actually the same problem. Do you think `@PublishedApi` is the way to go?

Reply:
Yes, I think it's ok to expose this function (`messagePrefix`) to public binary API.